backend/src/main/kotlin/com/hiper2d/auth/filter/JwtAuthenticationWebFilter.kt

```diff
@@ -0,0 +1,33 @@
+package com.hiper2d.auth.filter
+
+import org.springframework.security.authentication.ReactiveAuthenticationManager
+import org.springframework.security.web.server.authentication.AuthenticationWebFilter
+import org.springframework.security.web.server.authentication.ServerAuthenticationConverter
+import org.springframework.web.server.ServerWebExchange
+import org.springframework.web.server.WebFilterChain
+import org.springframework.web.server.adapter.DefaultServerWebExchange
+import reactor.core.publisher.Mono
+
+class JwtAuthenticationWebFilter(
+    authenticationManager: ReactiveAuthenticationManager,
+    authenticationTokenConverter: ServerAuthenticationConverter
+) : AuthenticationWebFilter(authenticationManager) {
+
+    init {
+        setServerAuthenticationConverter(authenticationTokenConverter)
+    }
+
+    override fun filter(exchange: ServerWebExchange, chain: WebFilterChain): Mono<Void> {
+        /*
```

Comment:
Maybe there's dependency issues or smth?

Reply:
Will debug it, hard to say